src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -3661,6 +3661,915 @@ class StubGenerator: public StubCodeGenerator {
+    // Copy elements [1..0] of v27 ({c,d}) into elements [3..2] of v16.
+    __ vslideup_vi(v16, v27, 2);                     // v16 = {d,c,_,_}
+    // Merge elements [1..0] of v26 ({a,b}) into elements [1..0] of v16
+    __ vmerge_vvm(v16, v16, v26);                    // v16 = {d,c,b,a}
+
+    // Save the hash
+    __ vse32_v(v17, state);
+    __ addi(state, state, -16);
+    __ vse32_v(v16, state);
+
+    __ leave();
+    __ ret();
+
+    return start;
+  }
+
+  // Arguments:
+  //
+  // Inputs:
+  //   c_rarg0   - byte[]  source+offset
+  //   c_rarg1   - int[]   SHA.state
+  //   c_rarg2   - int     offset
+  //   c_rarg3   - int     limit
+  //
+  address generate_sha512_implCompress(bool multi_block, const char *name) {
+    static const uint64_t round_consts[80] = {
+      0x428a2f98d728ae22l, 0x7137449123ef65cdl, 0xb5c0fbcfec4d3b2fl,
+      0xe9b5dba58189dbbcl, 0x3956c25bf348b538l, 0x59f111f1b605d019l,
+      0x923f82a4af194f9bl, 0xab1c5ed5da6d8118l, 0xd807aa98a3030242l,
+      0x12835b0145706fbel, 0x243185be4ee4b28cl, 0x550c7dc3d5ffb4e2l,
+      0x72be5d74f27b896fl, 0x80deb1fe3b1696b1l, 0x9bdc06a725c71235l,
+      0xc19bf174cf692694l, 0xe49b69c19ef14ad2l, 0xefbe4786384f25e3l,
+      0x0fc19dc68b8cd5b5l, 0x240ca1cc77ac9c65l, 0x2de92c6f592b0275l,
+      0x4a7484aa6ea6e483l, 0x5cb0a9dcbd41fbd4l, 0x76f988da831153b5l,
+      0x983e5152ee66dfabl, 0xa831c66d2db43210l, 0xb00327c898fb213fl,
+      0xbf597fc7beef0ee4l, 0xc6e00bf33da88fc2l, 0xd5a79147930aa725l,
+      0x06ca6351e003826fl, 0x142929670a0e6e70l, 0x27b70a8546d22ffcl,
+      0x2e1b21385c26c926l, 0x4d2c6dfc5ac42aedl, 0x53380d139d95b3dfl,
+      0x650a73548baf63del, 0x766a0abb3c77b2a8l, 0x81c2c92e47edaee6l,
+      0x92722c851482353bl, 0xa2bfe8a14cf10364l, 0xa81a664bbc423001l,
+      0xc24b8b70d0f89791l, 0xc76c51a30654be30l, 0xd192e819d6ef5218l,
+      0xd69906245565a910l, 0xf40e35855771202al, 0x106aa07032bbd1b8l,
+      0x19a4c116b8d2d0c8l, 0x1e376c085141ab53l, 0x2748774cdf8eeb99l,
+      0x34b0bcb5e19b48a8l, 0x391c0cb3c5c95a63l, 0x4ed8aa4ae3418acbl,
+      0x5b9cca4f7763e373l, 0x682e6ff3d6b2b8a3l, 0x748f82ee5defb2fcl,
+      0x78a5636f43172f60l, 0x84c87814a1f0ab72l, 0x8cc702081a6439ecl,
+      0x90befffa23631e28l, 0xa4506cebde82bde9l, 0xbef9a3f7b2c67915l,
+      0xc67178f2e372532bl, 0xca273eceea26619cl, 0xd186b8c721c0c207l,
+      0xeada7dd6cde0eb1el, 0xf57d4f7fee6ed178l, 0x06f067aa72176fbal,
+      0x0a637dc5a2c898a6l, 0x113f9804bef90dael, 0x1b710b35131c471bl,
+      0x28db77f523047d84l, 0x32caab7b40c72493l, 0x3c9ebe0a15c9bebcl,
+      0x431d67c49c100d4cl, 0x4cc5d4becb3e42b6l, 0x597f299cfc657e2al,
+      0x5fcb6fab3ad6faecl, 0x6c44198c4a475817l
+    };
+    __ align(CodeEntryAlignment);
+    StubCodeMark mark(this, "StubRoutines", name);
+    address start = __ pc();
+
+    Register buf   = c_rarg0;
+    Register state = c_rarg1;
+    Register ofs   = c_rarg2;
+    Register limit = c_rarg3;
+    Register consts = t0;
```

Comment:
Similar here. Please consider using `t1` instead for `consts`.

Reply:
Used t2 with spill.